### ISSUE TITLE
Join Cretonne and the codegen working group

### ISSUE THREAD

**Issue by johnrieth-zz** [nox/wg-codegen#3]
### I have raised this issue to discuss the possibility of the codegen working group focusing primarily on the development of [cretonne](https://github.com/cretonne/cretonne). I hope my reasoning for doing so and ideas about what the working group could do are helpful.
-------------------------------------------------------------------------------------------------------------------
I think the codegen working group's main goal should be the development of the [cretonne code generator](https://github.com/cretonne/cretonne). The cretonne code generator is primarily written in Rust  and a goal for the project is to be a [a full optimizing release-capable backend](https://github.com/cretonne/cretonne/blob/master/rustc.rst). Putting cretonne in front of the LLVM goals could help make the Rust compiler be written entirely in Rust. I am assuming that people who want to make the Rust compiler faster will want to write Rust instead of C++. Cretonne also seems to be more contributor friendly for Rustaceans of all skill levels LLVM.

**Ideas about how the working group could help cretonne**
- A website similar to llvm.org to be the central hub (cretonne-lang.org is available)
- Work to get sponsors to fund the development of cretonne
    - Work to get computer science researchers and students to help develop cretonne
- Make a team or teams similar to the [Rust teams](https://www.rust-lang.org/en-US/community.html) could help structure the development of cretonne 
    -  a team could focus on building [programming](http://tvmlang.org/) [languages](https://julialang.org/) [using cretonne](https://github.com/sunfishcode/simplejit-demo) 
    - One team could work with the [formal verification working group](https://rust-lang-nursery.github.io/wg-verification/) and other working groups.

Thank you.

**Comment by nox** [comment]
This is an interesting idea, but I feel like Cretonne should be its own working group.

I created the codegen WG specifically because I encountered real codegen problems in Servo and Stylo, with the existing codebase. There are various improvements that have been made already and will continue to be made in both the Rust codebase and LLVM upstream (thanks to @nikic, who I will never thank enough), and while Cretonne would be *super* nice to have long-term, I don't think this is something *this* WG should focus on, at least not for now.

Cretonne people are free to hang in `#wg-codegen` on <strike>Freenode</strike> the Mozilla IRC network (`ETOOMANYIRCNETWORKS`) and share ideas with us though.

*Edit: fixed the IRC network.*
